node/lib/cmd/cherry_pick.js

```diff
@@ -73,29 +88,56 @@ exports.configureParser = function (parser) {
  * @param {String[]} args.commits
  */
 exports.executeableSubcommand = co.wrap(function *(args) {
-    // TODO:
-    // - abort
-    // - continue
-
     const colors = require("colors");
 
-    const CherryPick      = require("../util/cherrypick");
+    const CherryPickUtil  = require("../util/cherry_pick_util");
     const GitUtil         = require("../util/git_util");
+    const UserError       = require("../util/user_error");
 
     const repo = yield GitUtil.getCurrentRepo();
 
+    if (args.continue + args.abort > 1) {
+        throw new UserError("Cannot use continue and abort together.");
+    }
+
+    if (args.continue) {
+        if (0 !== args.commits.length) {
+            throw new UserError("Cannot specify commits with '--continue'.");
+        }
+        const result = yield CherryPickUtil.continue(repo);
+        if (null !== result.errorMessage) {
+            throw new UserError(result.errorMessage);
+        }
+        return;                                                       // RETURN
+    }
+
+    if (args.abort) {
+        if (0 !== args.commits.length) {
+            throw new UserError("Cannot specify commits with '--abort'.");
+        }
+        yield CherryPickUtil.abort(repo);
+        return;                                                       // RETURN
+    }
+
+    if (0 === args.commits.length) {
+        throw new UserError("No commits to cherry-pick.");
+    }
+
     for (let i = 0; i < args.commits.length; ++i) {
         let commitish = args.commits[i];
-        let result = yield GitUtil.resolveCommitish(repo, commitish);
-        if (null === result) {
-            console.error(`Could not resolve ${colors.red(commitish)} to a \
-commit.`);
-            process.exit(-1);
+        const annotated = yield GitUtil.resolveCommitish(repo, commitish);
+        if (null === annotated) {
+            throw new UserError(`\
+Could not resolve ${colors.red(commitish)} to a commit.`);
         }
         else {
-            console.log(`Cherry-picking commit ${colors.green(result.id())}.`);
-            let commit = yield repo.getCommit(result.id());
-            yield CherryPick.cherryPick(repo, commit);
+            const id = annotated.id();
+            console.log(`Cherry-picking commit ${colors.green(id.tostrS())}.`);
+            const  commit = yield repo.getCommit(id);
+            const result = yield CherryPickUtil.cherryPick(repo, commit);
+            if (null !== result.errorMessage) {
```

Comment:
Do we need to store this and the remaining  commits in .git/sequencer/todo?

Reply:
State management is done in `CherryPickFileUtil`. I'm not using the sequencer yet. There's no support in libgit2 for it that I can see, and I'd actually prefer that norma git commands not see my state (so we don't run into problems like what we have today with people running `git rebase --abort` on git-meta rebases.

However, I would like to replicate the sequencer technique and support, e.g., `rebase -I` at some point -- but I had to draw a line on this change at some point.